packages/camera/camera_android_camerax/lib/src/camera_state.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart';
+
+import 'camera_state_error.dart';
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+
+/// The state of a camera.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/CameraState.
+class CameraState extends JavaObject {
+  /// Constructs a [CameraState] that is not automatically attached to a native object.
+  CameraState.detached(
+      {super.binaryMessenger,
+      super.instanceManager,
+      required this.type,
+      this.error})
+      : super.detached();
+
+  /// The type of state that the camera is in.
+  final CameraStateType type;
+
+  /// The error that the camera has encountered, if any.
```

Comment:
The most recent error? Does it persist across app restarts? Is this a mirrored value from android code? Can the value ever be cleared?

Reply:
You get access to `CameraState` through a `LiveData<CameraState>` so this state is from a snapshot of a point in time. It follows that this error is also representative of that snapshot. This is wrapping [`getError()`](https://developer.android.com/reference/androidx/camera/core/CameraState#getError()), so no, it can't be changed.

Reply:
Ah ok, so what I am hearing is that the caller has more context than I do when accessing this data. In that case consider updating the dart doc to indicate the snapshot like status of this error.